### DISCUSSION TITLE
How to run JobFunnel without installation for development purposes?

### DISCUSSION THREAD

**Discussion by anoduck** [PaulMcInnis/JobFunnel#162] [Q&A] [answered]
Perhaps I just tried the wrong file, but I am wondering if there is a way to run JobFunnel without having to run the installation command `python setup.py install` every time a change is made to the source code? Is there not a means by which one could just run `python main.py $(args)`? I am working on updating the bs4 selectors, and would like to test out acceptability, but this is very tedious when I have to reinstall the script after every change?

Also, not sure how to generate debug information to know if the selectors were correct or not.

**Comment by PaulMcInnis** [comment, marked as answer]
yup you can run without installing, and you can also install in-place, from the source

**Reply by anoduck** [reply]
@PaulMcInnis Weird, must be some Jedi trick to this, because when I attempt to run `python __main__.py -c ../my_settings.yaml` I receive an error concerning relative import with no known package.

```bash
Traceback (most recent call last):
  File "/home/user/jobfunnel/jobfunnel/__main__.py", line 6, in <module>
    from .backend.jobfunnel import JobFunnel
ImportError: attempted relative import with no known parent package
```

**Reply by anoduck** [reply]
I got it resolved.
